syncd/FlexCounter.cpp

```diff
@@ -995,10 +1080,41 @@ class AttrContext : public CounterContext<AttrType>
                 values.emplace_back(meta->attridname, sai_serialize_attr_value(*meta, attrs[i]));
             }
             countersTable.set(sai_serialize_object_id(vid), values, "");
+            if (switchType == "voq")
+            {
+               // Chassis based system
+               appDbCountersTable->set(sai_serialize_object_id(vid), values, "");
+            }
         }
     }
 };
 
+void FlexCounter::getCfgSwitchType(
+         _In_ std::string &switch_type)
+{
+   try
+   {
+      if (!m_cfgDeviceMetaDataTable->hget("localhost", "switch_type", switch_type))
+      {
+         //Switch type is not configured. Consider it default = "switch" (regular switch)
+         switch_type = "switch";
+      }
+   }
+   catch(const std::system_error& e)
+   {
+      SWSS_LOG_ERROR("System error: %s", e.what());
+      switch_type = "switch";
+   }
+
+   if (switch_type != "voq" && switch_type != "fabric" && switch_type != "chassis-packet" && switch_type != "switch" && switch_type != "dpu")
+   {
+      SWSS_LOG_ERROR("Invalid switch type %s configured", switch_type.c_str());
```

Comment:
this seems really fishy here, why you need to check all those variables? this should be more unified?

Reply:
Copying this function from https://github.com/sonic-net/sonic-swss/blob/master/orchagent/main.cpp#L160